stage1/init/pod_test.go

```diff
@@ -54,3 +62,95 @@ func TestQuoteExec(t *testing.T) {
 		}
 	}
 }
+
+// TestAppToNspawnArgsOverridesImageManifestReadOnly tests
+// that the ImageManifest's `readOnly` volume setting will be
+// overrided by PodManifest.
+func TestAppToNspawnArgsOverridesImageManifestReadOnly(t *testing.T) {
+	falseVar := false
+	trueVar := true
+	tests := []struct {
+		imageManifestVolumeReadOnly bool
+		podManifestVolumeReadOnly   *bool
+		expectReadOnly              bool
+	}{
+		{
+			false,
+			nil,
+			false,
+		},
+		{
+			false,
+			&falseVar,
+			false,
+		},
+		{
+			false,
+			&trueVar,
+			true,
+		},
+		{
+			true,
+			nil,
+			true,
+		},
+		{
+			true,
+			&falseVar,
+			false,
+		},
+		{
+			true,
+			&trueVar,
+			true,
+		},
+	}
+
+	for i, tt := range tests {
+		podManifest := &schema.PodManifest{
+			Volumes: []types.Volume{
+				{
+					Name:     "foo-mount",
+					Kind:     "host",
+					Source:   "/host/foo",
+					ReadOnly: tt.podManifestVolumeReadOnly,
+				},
+			},
+		}
+		appManifest := &schema.RuntimeApp{
+			Mounts: []schema.Mount{
+				{
+					Volume: "foo-mount",
+					Path:   "/app/foo",
+				},
+			},
+			App: &types.App{
+				Exec:  []string{"/bin/foo"},
+				User:  "0",
+				Group: "0",
+				MountPoints: []types.MountPoint{
+					{
+						Name:     "foo-mount",
+						Path:     "/app/foo",
+						ReadOnly: tt.imageManifestVolumeReadOnly,
+					},
+				},
+			},
+		}
+
+		tmpDir, err := ioutil.TempDir("", tstprefix)
+		if err != nil {
+			t.Errorf("error creating tempdir: %v", err)
+		}
+		defer os.RemoveAll(tmpDir)
+
+		p := &Pod{Manifest: podManifest, Root: tstprefix}
+		output, err := p.appToNspawnArgs(appManifest)
+		if err != nil {
+			t.Errorf("#%d: unexpected error: `%v`", i, err)
+		}
+		if ro := strings.HasPrefix(output[0], "--bind-ro"); ro != tt.expectReadOnly {
```

Comment:
How about actually checking the whole array if a regex like `^--(bind(-ro)?)=/host/foo:.*/app/foo$` matches in any of them and if the first submatch is `bind` or `bind-ro` depening on `tt.expectReadOnly`?